codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -346,19 +375,16 @@ public void headersWriteForUnknownStreamShouldCreateStream() throws Exception {
 
     @Test
     public void headersWriteShouldCreateHalfClosedStream() throws Exception {
-        int streamId = 5;
-        when(stream.id()).thenReturn(streamId);
-        when(stream.state()).thenReturn(IDLE);
-        mockFutureAddListener(true);
-        when(local.createStream(eq(streamId))).thenReturn(stream);
-        when(writer.writeHeaders(eq(ctx), eq(streamId), eq(EmptyHttp2Headers.INSTANCE), eq(0),
-                eq(DEFAULT_PRIORITY_WEIGHT), eq(false), eq(0), eq(true), eq(promise))).thenReturn(future);
-        encoder.writeHeaders(ctx, streamId, EmptyHttp2Headers.INSTANCE, 0, true, promise);
-        verify(local).createStream(eq(streamId));
-        verify(stream).open(eq(true));
-        // Trigger the write and mark the promise successful to trigger listeners
-        payloadCaptor.getValue().write(0);
-        promise.trySuccess();
+        Http2Connection connection = new DefaultHttp2Connection(true);
```

Comment:
You mentioned [here](https://github.com/netty/netty/pull/3456#issuecomment-76834859) that you wanted to test some  flow controller behavior and that is why we couldn't use the mocks?  Could we break the test apart to test the `DefaultHttp2ConnectionEncoder` here (also update the mocks to simulate the expected behavior of the flow controller), and then add/modify tests in `DefaultHttp2RemoteFlowControllerTest` to test the flow controller's responsibilties?